sdk/keyvault/azure-security-keyvault-administration/README.md

```diff
@@ -130,6 +130,137 @@ To update the value of any of the the available settings, we will call the Updat
+
+```cpp Snippet:SampleAdministration2CreateClient
+  // create client
+  BackupRestoreClient client(std::getenv("AZURE_KEYVAULT_HSM_URL"), credential);
+```
+## Create the SasTokenParameter
+
+Since these operations require a blob storage for the backup/restore operations, a SAS token is required for the connection between the services(Key Vault and Storage).  
+
+In this sample we rely on a couple of extra environment variables. 
+
+```cpp Snippet:SampleAdministration2CreateSASParam
+ SasTokenParameter sasTokenParameter;
+ // the backup/restore needs a SAS token to access the storage account
+ sasTokenParameter.Token
+     = Azure::Core::_internal::Environment::GetVariable("AZURE_KEYVAULT_BACKUP_TOKEN");
+ // the backup/restore needs a url to a blob storage resource
+ sasTokenParameter.StorageResourceUri
+     = Azure::Core::_internal::Environment::GetVariable("AZURE_KEYVAULT_BACKUP_URL");
+```
+
+## The Backup operation 
+
+Since this is a long running operation the service provides endpoints to determine the status while the opperation is running. 
+
+### Starting the backup operation
+
+```cpp Snippet:SampleAdministration2StartBackup
+// Create a full backup using a user-provided SAS token to an Azure blob storage container.
+auto backupResponse = client.FullBackup(sasTokenParameter);
+
+std::cout << "Backup Job Id: " << backupResponse.Value.JobId << std::endl
+          << "Backup Status: " << backupResponse.Value.Status << std::endl;
+```
+
+### Backup status check 
+
+In order to determine the status of the operation we need to call the corresponding endpoint.
+
+```cpp Snippet:SampleAdministration2StatusBackup
+// Get the status of the backup operation.
+auto backupStatus = client.FullBackupStatus(backupResponse.Value.JobId);
+
+// Wait for the backup to complete.
+while (backupStatus.Value.Status == "InProgress")
+{
+  std::this_thread::sleep_for(5s);
+  backupStatus = client.FullBackupStatus(backupStatus.Value.JobId);
+}
+std::cout << "Backup Job Id: " << backupStatus.Value.JobId << std::endl
+          << "Backup Status: " << backupStatus.Value.Status << std::endl;
+```
+
+## The FullRestore operation
+
+Similar to the backup operation after we initialize the operation we can check the status. 
+
+### Starting the restore operation 
+
+the restore operation requires a folder where a backup was previously performed along side the SAS token parameter. 
+```cpp Snippet:SampleAdministration2FullRestoreStart
+// Restore the full backup using a user-provided SAS token to an Azure blob storage container.
+RestoreOperationParameters restoreBlobDetails;
+restoreBlobDetails.SasTokenParameters = sasTokenParameter;
+restoreBlobDetails.FolderToRestore = "folder name here";
+std::cout << "Folder to restore: " << restoreBlobDetails.FolderToRestore << std::endl;
+auto restoreResponse = client.FullRestore(restoreBlobDetails);
```

Comment:
Are the folder to restore and sas token required parameters or are they optional parameters? If they're required parameters, they should be explicitly called out in the FullRestore method signature. 

If they are optional parameters, do you distinguish between the empty string and "none"?

Reply:
fixed ti to match the .Net API